### DISCUSSION TITLE
Is there any way to subscribe to a collection and get all records in the results?

### DISCUSSION THREAD

**Discussion by ryu-man** [pocketbase/pocketbase#1984] [Q&A] [answered]
Hello, first of all, I wanna say thanks for this amazing solution.

if I want to subscribe to a collection and get all records as an array in the results, how can I achieve that?

**Comment by ganigeorgiev** [comment, marked as answer]
The realtime subscriptions currently react only to record changes (create/update/delete), aka. you can't fetch all records on `subscribe("*", ...)`.

If you want to fetch all records you can call manually `pb.collection("example").getFullList()` (_under the hood it will send batched `getList()` requests until all records are fetched_).

**Comment by ryu-man** [comment]
That is what I am doing right now

**Comment by ryu-man** [comment]
We really need subscriptions on collections and queries just like in firebase firestore
